astropy/visualization/wcsaxes/patches.py

```diff
@@ -81,9 +84,20 @@ class SphericalCircle(Polygon):
 
     def __init__(self, center, radius, resolution=100, vertex_unit=u.degree, **kwargs):
 
-        # Extract longitude/latitude, either from a tuple of two quantities, or
-        # a single 2-element Quantity.
-        longitude, latitude = center
+        # Extract longitude/latitude, either from a SkyCoord object, or
+        # from a tuple of two quantities or a single 2-element Quantity.
+        # The SkyCoord is converted to SphericalRepresentation, if not already.
+        if isinstance(center, SkyCoord):
+            rep_type = center.representation_type
+            if rep_type != SphericalRepresentation:
+                warnings.warn(f'Received center of representation type {rep_type} '
+                              'will be converted to Spherical Representation ',
+                              UserWarning)
+                center.representation_type = 'spherical'
```

Comment:
This has the side effect that the `SkyCoord` the user has provided is converted to spherical coordinates. We do want to obtain the longitude and latitude, but any conversion should stay inside the function and not modify the objects that were passed as inputs. Accessing `center.spherical.lat` and `center.spherical.lon` would avoid that problem.

Reply:
You could use `center.spherical.lat` and `center.spherical.lon` even if the `SkyCoord` already is in spherical coordinates, and then you wouldn't need `center.representation_component_names` at all.